clients/cli/cmd/internal/pull.go

```diff
@@ -169,33 +173,128 @@ func (target *Target) DownloadAndWriteToFile(client *phrase.APIClient, localeFil
 		fmt.Fprintln(os.Stderr, "FormatOptions", localVarOptionals.FormatOptions)
 	}
 
-	file, response, err := client.LocalesApi.LocaleDownload(Auth, target.ProjectID, localeFile.ID, &localVarOptionals)
-	if err != nil {
-		if response.Rate.Remaining == 0 {
-			waitForRateLimit(response.Rate)
-			file, _, err = client.LocalesApi.LocaleDownload(Auth, target.ProjectID, localeFile.ID, &localVarOptionals)
-			if err != nil {
+	if async {
+		localeDownloadCreateParams := asyncDownloadParams(localVarOptionals)
+
+		localVarOptionals := phrase.LocaleDownloadCreateOpts{}
+		if Debug {
+			fmt.Fprintln(os.Stderr, "Initiating async download...")
+		}
+		asyncDownload, _, _ := client.LocaleDownloadsApi.LocaleDownloadCreate(Auth, target.ProjectID, localeFile.ID, localeDownloadCreateParams, &localVarOptionals)
+
+		for asyncDownload.Status == "processing" {
+			if Debug {
+				fmt.Fprintln(os.Stderr, "Waiting for the files to be exported...")
```

Comment:
Was thinking if we can have some sort of function roughly like

```
func debugLog(debugMessage string) {
  if Debug {
    fmt.Fprintln(os.Stderr, debugMessage)
  }
}
```

And use it in place of conditionally logging debug messages? I understand this is existing way of logging debug messages but I was just thinking as a starting point this can to some extent do code clean up.

Reply:
Or we can have it as a future improvement, where this Debug function can be exposed by some other package or same file within the same package and used app wide.